server/controllers/people/timelineController.ts

```diff
@@ -28,16 +28,16 @@ export default class TimelineController {
 
       if (!crn?.trim()) {
         addErrorMessageToFlash(req, 'You must enter a CRN', 'crn')
-        res.redirect(paths.timeline.find({}))
+        return res.redirect(paths.timeline.find({}))
       }
 
       try {
         const timeline = await this.personService.getTimeline(req.user.token, crn.trim())
 
-        res.render('people/timeline/show', { timeline, crn, pageHeading: `Timeline for ${crn}` })
+        return res.render('people/timeline/show', { timeline, crn, pageHeading: `Timeline for ${crn}` })
       } catch (error) {
         crnErrorHandling(req, error, crn)
-        res.redirect(paths.timeline.find({}))
+        return res.redirect(paths.timeline.find({}))
```

Comment:
thanks Bala, do you think you could:

- add into the commit message an explanation of the problem and how this fixes it?
- include a unit test to document this in the code and prevent regression?

Reply:
@edavey 
add into the commit message an explanation of the problem and how this fixes it?
updated the commit message
include a unit test to document this in the code and prevent regression?
502 error gateway if submitting person timeline multiple times with no CRN
Person timeline crn search gives error on second time submit 
the issue is on TimelineController redirect with out return and so it is updating the response headers after repose send to client
Normal unit test is already there and
is not possible to write a unit test especially for this scenario